src/Microsoft.ML.Data/Prediction/Calibrator.cs

```diff
@@ -430,8 +432,13 @@ private static VersionInfo GetVersionInfoBulk()
                 loaderAssemblyName: typeof(ValueMapperCalibratedModelParameters<TSubModel, TCalibrator>).Assembly.FullName);
         }
 
-        private ValueMapperCalibratedModelParameters(IHostEnvironment env, ModelLoadContext ctx)
-            : base(env, RegistrationName, GetPredictor(env, ctx), GetCalibrator(env, ctx))
+        //private ValueMapperCalibratedModelParameters(IHostEnvironment env, ModelLoadContext ctx)
+        //    : base(env, RegistrationName, GetPredictor(env, ctx), GetCalibrator(env, ctx))
```

Comment:
@eerhardt @yaeldekel I have a question about this. As I pointed out in point 1) of the problems on issue #4292 , the Create method of ParameterMixingCalibratedModelParameters (PMCMP) wasn't called when loading that model, and instead its constructor with parameters (IHostEnvironment env, ModelLoadContext ctx) was called directly without calling the Create Method. This happens because the [CreateInstanceCoreMethod ](https://github.com/dotnet/machinelearning/blob/7c067854b564275b0d6387ca59c0ec83e8fc91b9/src/Microsoft.ML.Core/ComponentModel/ComponentCatalog.cs#L197) looks for a constructor before a create method. To solve that I created an static version of `PMCMP` with a create method that calls the create method of generic `PMCMP<,>`

This problem actually happens also in the other `CalibratedModelParameters `classes, but I have just realized that it is actually not necessary to create another class to bypass this. As I show here in `ValueMapperCalibratedModelParameters`, I can solve this problem by actually removing the constructor `VMCMP(env,ctx)` so that `Create(env, ctx)` of VMCMP actually gets called.

Removing this constructor seems safe to me, since it was only referenced by the `Create(env, ctx) ` method that was being ignored anyway, and which I've updated, and now doesn't call that constructor, but calls constructor `VMCMP(env, ctx, predictor, calibrator)`. I've done this also in the `FeatureWeightsCalibratedModelParameters` class. And all the tests pass as expected.

So I just wanted to double check with you if it's safe to remove that constructor (I don't know if it might be, perhaps, called through reflection somewhere else, or something).

If it's safe, then I will also remove it in PMCMP and creating the static PMCMP wouldn't be necessary either.

Thanks.

Reply:
My understanding is that it is safe to remove the constructor as well - because we have the `Create` method instead in its place. I _think_ the only place that looks for it is in the `ComponentCatalog`. Doing a quick search for `ConstructorInfo` across the code, that appears to be the case.

So I would go ahead and remove the constructor, and use the static `Create` method instead.

Reply:
Ok. So I've deleted the constructor(env,ctx) from PMCMP, VMCMP and FWCMP. I've also deleted the non-generic PMCMP class that I had previously added on this PR, since it's no longer necessary.